design-proposals/template-generation.md

```diff
@@ -0,0 +1,127 @@
+# Overview
+
+This design defines an approach to create Virtual Machine templates starting from an existing VM.
+
+## Motivation
+
+Creating a 'blueprint' of a running VM to further tweak and use as a basis golden VM template is a common operation, also it is a tool that can be leveraged by UIs to improve the UX of the project.
+
+## Goals
+
+* Create a generic template of a running VM.
+
+## Non Goals
+
+* Designing a VM-cloning mechanism, which copies disk data or create a snapshot of the live memory of a VM.
+
+## User Stories
+
+* As a cluster user, I want to create a re-usable VM spec to use as the basis for my VMs workloads starting from a running VM.
+* As a cluster user, I want to create different workloads based on the same VM image by just tweaking some template parameters.
+
```

Comment:
In the template, there's a `repos` section right about there. I think it would make sense to specify where the work would happen (ssp-operator?).

Reply:
Added!